enterprise/server/api/api_server.go

```diff
@@ -513,6 +507,11 @@ func (s *APIServer) ExecuteWorkflow(ctx context.Context, req *apipb.ExecuteWorkf
 	}
 	rsp, err := wfs.ExecuteWorkflow(ctx, r)
 	if err != nil {
+		if status.IsNotFoundError(err) {
```

Comment:
Just confirming, all that's changed here is that this failure mode is handled by the service instead of checking before making the request, yeah?

Reply:
Yeah. I realized it was making an unnecessary additional DB read before, so moved things around to eliminate it